src/library/Uno.Material/MaterialTheme.cs

```diff
@@ -11,196 +9,25 @@
-		/// </summary>
-		public ResourceDictionary FontOverrideDictionary
-		{
-			get => (ResourceDictionary)GetValue(FontOverrideDictionaryProperty);
-			set => SetValue(FontOverrideDictionaryProperty, value);
-		}
-
-		public static DependencyProperty FontOverrideDictionaryProperty { get; } =
-			DependencyProperty.Register(
-				nameof(FontOverrideDictionary),
-				typeof(ResourceDictionary),
-				typeof(MaterialTheme),
-				new PropertyMetadata(null, OnFontOverrideChanged));
-
-		private static void OnFontOverrideChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
-		{
-			if (d is MaterialTheme { _isFontOverrideMuted: false } theme)
-			{
-				theme.UpdateSource();
-			}
-		}
-		#endregion
-
-		#region ColorOverrideDictionary (DP)
-		/// <summary>
-		/// (Optional) Gets or sets a <see cref="ResourceDictionary"/> containing overrides for the default Uno.Material <see cref="Color"/> resources
-		/// </summary>
-		/// <remarks>The overrides set here should be re-defining the <see cref="Color"/> resources used by Uno.Material, not the <see cref="SolidColorBrush"/> resources</remarks>
-		public ResourceDictionary ColorOverrideDictionary
-		{
-			get => (ResourceDictionary)GetValue(ColorOverrideDictionaryProperty);
-			set => SetValue(ColorOverrideDictionaryProperty, value);
-		}
-
-		public static DependencyProperty ColorOverrideDictionaryProperty { get; } =
-			DependencyProperty.Register(
-				nameof(ColorOverrideDictionary),
-				typeof(ResourceDictionary),
-				typeof(MaterialTheme),
-				new PropertyMetadata(null, OnColorOverrideChanged));
-
-		private static void OnColorOverrideChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
-		{
-			if (d is MaterialTheme { _isColorOverrideMuted: false } theme)
-			{
-				theme.UpdateSource();
-			}
-		}
-		#endregion
-
-		public MaterialTheme() : this(colorOverride: null, fontOverride: null)
-		{
-			
-		}
+		public MaterialTheme()
+		{ }
 
 		public MaterialTheme(ResourceDictionary colorOverride = null, ResourceDictionary fontOverride = null)
-		{
-			if (colorOverride is { })
-			{
-				SetColorOverrideSilently(colorOverride);
-			}
-
-			if (fontOverride is { })
-			{
-				SetFontOverrideSilently(fontOverride);
-			}
-
-			UpdateSource();
-		}
-
-		private void SetColorOverrideSilently(ResourceDictionary colorOverride)
-		{
-			try
-			{
-				_isColorOverrideMuted = true;
-				ColorOverrideDictionary = colorOverride;
-			}
-			finally
-			{
-				_isColorOverrideMuted = false;
-			}
-		}
-
-		private void SetFontOverrideSilently(ResourceDictionary fontOverride)
-		{
-			try
-			{
-				_isFontOverrideMuted = true;
-				FontOverrideDictionary = fontOverride;
-			}
-			finally
-			{
-				_isFontOverrideMuted = false;
-			}
-		}
+			: base(colorOverride, fontOverride)
+		{ }
 
-		private void UpdateSource()
+		protected override (ResourceDictionary mergedPages, ResourceDictionary[] resources) GenerateSpecificResources()
```

Comment:
I just planned to have an API that will be good in different scenarios, if in the future we've more resources than `fonts` or for Fluent we've more resources this will be able to handle already. IF you think that case will hardly happen, I'll be happy to make it return just one `ResourceDictionary`

Reply:
@Xiaoy312 / @agneszitte what do you think? I think we should just provide a spot here where the concrete theme class can provide a ResourceDictionary to be included in the merge dicts?

Reply:
we could have this return a single res-dict instead
if the order of insertion matter, we can pass in the colors res-dict, so it can be appended before the rest

Reply:
Yeah I believe a single res-dict would work and it shouldn't need to pass the colors as those will be added as a merged-dict to the returned res-dict